modules/csg/csg.cpp

```diff
@@ -1024,9 +1024,6 @@ int CSGBrushOperation::MeshMerge::_create_bvh(BVH *p_bvh, BVH **p_bb, int p_from
 			p_bb[p_from + i]->next = p_bb[p_from + i + 1] - p_bvh;
 		}
 		return p_bb[p_from] - p_bvh;
-	} else if (p_size == 0) {
```

Comment:
I guess that code should likely be moved up before comparing with `BVH_LIMIT`. That would change the logic, but it does look like that was the intention.

Reply:
CC @reduz

Reply:
For this one @reduz says:

> looks ok, i guess it never reaches zero [...], so that's why it doesn't fail

So the proposed fix should be good/safe.